client/firecracker_client_factory_test.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"github.com/stretchr/testify/assert"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestNewHttpClientForUNIXSocket(t *testing.T) {
+	done := make(chan bool)
+
+	expectedMessage := "PUT /logger HTTP/1.1\r\nHost: localhost\r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 0\r\nAccept: application/json\r\nContent-Type: application/json\r\nAccept-Encoding: gzip\r\n\r\n"
+	socketName := "testingUnixSocket.sock"
+	addr, _ := net.ResolveUnixAddr("unix", socketName)
+	listener, _ := net.ListenUnix("unix", addr)
+	defer listener.Close()
+
+	go func() {
+		conn, err := listener.AcceptUnix()
+		if err != nil {
+			t.Error(err)
+		}
+
+		for {
+			buf := make([]byte, 512)
+			nr, err := conn.Read(buf)
+			if err != nil {
+				return
+			}
+
+			data := string(buf[0:nr])
+			assert.Equal(t, expectedMessage, data, "expectedMessage received on socket is different than what was sent")
```

Comment:
can we just do a straight compare here to be consistent with the other tests?

```go
if e, a := expectedMessage, data; e != a {
    t.Errorf("expected message to be %q, but received %q", e, a)
}
```

Reply:
Sure! Happy to get rid of the assert if we're staying staying away from testify for now.

Reply:
For something like string comparison, testify's diff on errors is pretty useful.  I'd lean toward keeping this as `assert.Equal`.

Reply:
@samuelkarp - If that is how we want to move forward, we probably want to take some time and update the rest of the tests to use the testify package. I am more for less dependencies the better for an SDK.